### PR TITLE
redux refactor profile page

### PR DESCRIPTION
**Pull request by beetai**
galaxy brain sepand strikes again

### CODE REVIEW

File: frontend/src/redux/reducers/Profile.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      case 'courses':
+        changedStore.courses = action.isEditing;
+        break;
+      default:
+        break;
+    }
+    return changedStore;
+  }
+  return store;
+};
```
Comment (DSep) [inline frontend/src/redux/reducers/Profile.tsx]: Can we just get rid of this state?

File: frontend/src/components/ProtectedRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const { currentUser, component: Component, ...rest } = props;
+  // if (!currentUser) {
+  //   return (
+  //     <Route path='/login' component={Login}></Route>
+  //   );
+  // } else {
+  //   return (
+  //     <Component  />
+  //   );
+  // }
```
Comment (DSep) [inline frontend/src/components/ProtectedRoute.tsx]: just comments that can be removed

File: frontend/src/type-interfaces/FieldEditState.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface FieldEditState {
+  name: boolean,
+  email: boolean,
+  major: boolean,
+  courses: boolean,
+}
```
Comment (DSep) [inline frontend/src/type-interfaces/FieldEditState.tsx]: Cool, but possibly overkill? At least as something in the store. Could use in local state instead

File: frontend/src/redux/actions/Profile.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const setField = (field: String, isEditing: boolean) => ({ // TODO: switch string to something like enum?
+  type: 'SET_FIELD',
+  field,
+  isEditing,
+});
```
Comment (DSep) [inline frontend/src/redux/actions/Profile.tsx]: Let's figure out if we can avoid this being an action

File: frontend/src/pages/AccountForm.tsx [hunk 5]
```diff
@@ -7,184 +7,155 @@ import Button from '@material-ui/core/Button';
...
-      major: profileMajor,
-      courses: newProfile.courses,
-      id: '5f080b5a1ad963c82c07f475',
-    });
-    toggleMajor(!editMajor);
+    // eslint-disable-next-line no-underscore-dangle
+    updateUser(currentUser._id, changedFields);
+    setField('name', false);
+    setField('email', false);
+    setField('major', false);
```
Comment (DSep) [inline frontend/src/pages/AccountForm.tsx]: Let's try and do this locally

**Review by DSep — CHANGES_REQUESTED** [review]
Let's see if we can avoid having the editing flags in the redux store.

File: frontend/src/pages/AccountForm.tsx [hunk 6]
```diff
@@ -1,190 +1,145 @@
...
-      });
+    // axios.get('/users')
+    //   .then((response) => {
+    //     const user = response.data[0];
+    //     setName(user.name);
+    //     setEmail(user.email);
+    //     setMajor(user.major);
+    //     setCourses(user.courses);
+    //   }); // TODO: WE SHOULD NOT BE CALLING THE API TWICE, THIS IS TEMP
+
```
Comment (DSep) [inline frontend/src/pages/AccountForm.tsx]: More comments

File: frontend/src/redux/reducers/User.tsx [hunk 7]
```diff
@@ -17,3 +17,14 @@ export const getUsers = (store: any[] = [], action: any) => {
       return store;
   }
 };
+
+export const getCurrentUser = (store: any = {}, action: any) => { // TODO: temporary setup for single user
+  switch (action.type) {
+    case 'GET_USERS':
+      return action.users[0];
```
Comment (baconandchips) [inline frontend/src/redux/reducers/User.tsx:24]: Will this be changed in the future? It seems to currently grab the first user only. Otherwise LGTM

**Review by yqj2k — APPROVED** [review]

File: frontend/src/redux/reducers/User.tsx (hunk 7 above)
Comment (baconandchips) [reply frontend/src/redux/reducers/User.tsx:24]: Add(ed) to card, tbd; perhaps in future

File: frontend/src/redux/reducers/User.tsx [hunk 8]
```diff
@@ -17,3 +17,14 @@ export const getUsers = (store: any[] = [], action: any) => {
       return store;
   }
 };
+
+export const getCurrentUser = (store: any = {}, action: any) => { // TODO: temporary setup for single user
+  switch (action.type) {
+    case 'GET_USERS':
```
Comment (DSep) [inline frontend/src/redux/reducers/User.tsx]: Make into constant pls

**Review by DSep — CHANGES_REQUESTED** [review]

File: frontend/src/redux/actions/User.tsx [hunk 9]
```diff
@@ -17,13 +17,31 @@ export const getUsersSuccess = (users : any[]) => ({
   users,
 });
 
+export const updateUserSuccess = (user: any) => ({ // TODO set the user type????
+  type: 'UPDATE_USER',
```
Comment (baconandchips) [inline frontend/src/redux/actions/User.tsx]: constant as well

**Review by DSep — APPROVED** [review]
LGTM now

**Pull request merged**